src/Microsoft.ML.StandardTrainers/Standard/ModelStatistics.cs

```diff
@@ -16,249 +16,343 @@
 using Microsoft.ML.Trainers;
 
 // This is for deserialization from a model repository.
-[assembly: LoadableClass(typeof(LinearModelStatistics), null, typeof(SignatureLoadModel),
+[assembly: LoadableClass(typeof(ModelStatisticsBase), typeof(LinearModelParameterStatistics), null, typeof(SignatureLoadModel),
     "Linear Model Statistics",
-    LinearModelStatistics.LoaderSignature)]
+    LinearModelParameterStatistics.LoaderSignature)]
+
+// This is for deserialization from a model repository.
+[assembly: LoadableClass(typeof(ModelStatisticsBase), typeof(ModelStatisticsBase), null, typeof(SignatureLoadModel),
+    "Model Statistics",
+   ModelStatisticsBase.LoaderSignature)]
 
 namespace Microsoft.ML.Trainers
 {
     /// <summary>
     /// Represents a coefficient statistics object.
     /// </summary>
-    public readonly struct CoefficientStatistics
+    public sealed class CoefficientStatistics
     {
-        public readonly string Name;
         public readonly float Estimate;
         public readonly float StandardError;
         public readonly float ZScore;
         public readonly float PValue;
 
-        internal CoefficientStatistics(string name, float estimate, float stdError, float zScore, float pValue)
+        internal CoefficientStatistics(float estimate, float stdError, float zScore, float pValue)
         {
-            Contracts.AssertNonEmpty(name);
-            Name = name;
             Estimate = estimate;
             StandardError = stdError;
             ZScore = zScore;
             PValue = pValue;
         }
     }
 
-    // REVIEW: Make this class a loadable class and implement ICanSaveModel.
     // REVIEW: Reconcile with the stats in OLS learner.
     /// <summary>
     /// The statistics for linear predictor.
     /// </summary>
-    public sealed class LinearModelStatistics : ICanSaveModel
+    public class ModelStatisticsBase : ICanSaveModel
```

Comment:
>public class [](start = 4, length = 12)

Is this intended to be an abstract class?

Reply:
no, i still need to instantiate it.

---
In reply to: [265254412](https://github.com/dotnet/machinelearning/pull/2048#discussion_r265254412) [](ancestors = 265254412)

Reply:
Who instantiates it?

---
In reply to: [265259755](https://github.com/dotnet/machinelearning/pull/2048#discussion_r265259755) [](ancestors = 265259755,265254412)

Reply:
Agree with Tom.

---
In reply to: [265277153](https://github.com/dotnet/machinelearning/pull/2048#discussion_r265277153) [](ancestors = 265277153,265259755,265254412)